faucet.go

```diff
@@ -708,3 +724,48 @@ func cleanAndExpandPath(path string) string {
 	// but the variables can still be expanded via POSIX-style $VARIABLE.
 	return filepath.Clean(os.ExpandEnv(path))
 }
+
+// generateInvoice is a hybrid http.Handler that handles: the validation of the
+// generate invoice form, rendering errors to the form, and finally generating
+// invoice if all the parameters check out.
+func (l *lightningFaucet) generateInvoice(homeTemplate *template.Template,
+	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
+
+	if err := r.ParseForm(); err != nil {
```

Comment:
Probably best to add a hard check at the start to limit the number of generated invoices per minute, since this can be pretty easily DoS'd.

The easiest check to add is probably to just store in the server the timestamp of the last generated invoice and limit a timeout of 60 seconds between invoices.